### DISCUSSION TITLE
NixOS package

### DISCUSSION THREAD

**Discussion by MyGitHubBlueberry** [eylles/pywal16#41] [Q&A]
Hi guys. I couldn't find nixos package for pywal16 and original pywal is not generating themes for hyprland(
So I will package it myself locally and paste package code there. Maybe it will help to make your official nixOS package eventually.

**Comment by MyGitHubBlueberry** [comment]
Something like that, I guess...
```
{ pkgs }:

pkgs.stdenv.mkDerivation {
    name = "pywal-sixteen"

    src = pkgs.fetchurl {
      url = "https://github.com/eylles/pywal16/releases/download/3.5.3.zip";
      hash = "0mh3cjl7fajwiwyimc5pjz9ib0dnv9dprv0wzl3ji0a35npjcfhn";
    };

    dontUnpack = true;

    installPhase = ''
        mkdir -p $out
        ${pkgs.unzip}/bin/unzip $src -d $out/
    '';
};
```

**Reply by eylles** [reply]
nice, if you can get it to be added to nix packages that would be great, idk the process tbh and all i can do myself is keep maintaining this and providing the pypi package, maybe a debian package since i got some experience packaging for debian but only that, so it is always good to have people eager to collaborate packaging for their distros.

**Reply by MyGitHubBlueberry** [reply]
Btw, above code, I wrote wouldn't work)

**Comment by MyGitHubBlueberry** [comment]
I made a package 
https://github.com/NixOS/nixpkgs/pull/312666

Update:
Solving issues in it.

**Reply by MyGitHubBlueberry** [reply]
It's not accepted yet, but the base is done already

**Comment by MyGitHubBlueberry** [comment]
It's working, but i'm having base 8 colors.
![image](https://github.com/eylles/pywal16/assets/105305430/31446f54-e438-4873-b865-c4f548555ac3)

It might be wpgtk, which uses pywal at back-end.

**Reply by MyGitHubBlueberry** [reply]
My bad...
![image](https://github.com/eylles/pywal16/assets/105305430/2bb1e486-e6dc-4c9c-a0ed-f9907de3645c)

**Reply by eylles** [reply]
Eh don't worry, i should probably be more clear about the default behaviour somewhere but haven't figured where.

**Reply by MyGitHubBlueberry** [reply]
maybe adding `#Usage` section to the repo `read.md` would work

**Comment by MyGitHubBlueberry** [comment]
I opened pull again.
https://github.com/NixOS/nixpkgs/pull/312684

**Reply by MyGitHubBlueberry** [reply]
![image](https://github.com/eylles/pywal16/assets/105305430/d0a4fa9b-9c00-483b-a3db-fd4c92acb473)

**Reply by MyGitHubBlueberry** [reply]
first message at the pull page says following
![image](https://github.com/eylles/pywal16/assets/105305430/caf1a929-ff2e-4082-8bcc-f6569f7f40dc)
So maybe you also could press a :+1: ? :)

**Reply by eylles** [reply]
Let's hope it goes through

**Reply by MyGitHubBlueberry** [reply]
![image](https://github.com/eylles/pywal16/assets/105305430/df943a87-1801-41d8-99d8-3294e66a64b5)

**Reply by MyGitHubBlueberry** [reply]
It still pending a review from the same guy. He is 3 days offline already :)

**Reply by MyGitHubBlueberry** [reply]
Btw, all requirements are met, request is still not pulled...
